modules/oe_theme_content_event/src/Plugin/ExtraField/Display/RegistrationButtonExtraField.php

```diff
@@ -99,12 +100,22 @@ public function viewElements(ContentEntityInterface $entity) {
     if ($event->isRegistrationPeriodYetToCome($this->requestDateTime)) {
       $datetime_start = $event->getRegistrationStartDate();
       $datetime_end = $event->getRegistrationEndDate();
-      $date_diff = $this->dateFormatter->formatDiff($this->requestTime, $datetime_start->getTimestamp());
-      $build['#description'] = $this->t('Registration will open in @time_left. You can register from @start_date, until @end_date.', [
-        '@time_left' => $date_diff,
-        '@start_date' => $this->dateFormatter->format($datetime_start->getTimestamp(), 'oe_event_date_hour'),
-        '@end_date' => $this->dateFormatter->format($datetime_end->getTimestamp(), 'oe_event_date_hour'),
-      ]);
+      $request_datetime = DrupalDateTime::createFromTimestamp($this->requestTime);
+      $date_diff = $request_datetime->diff($datetime_start);
+      if ($date_diff->days === 0) {
+        $build['#description'] = $this->t('Registration will close today, @end_date.', [
```

Comment:
I don't think this message is very useful as it is. It will be shown to a user when it arrives at an event the day the registration opens but at an earlier hour. Therefore I think we should inform the user when they can expect to be able to register: "Registration will open today at @hour." sounds more appropriate.

Reply:
Yes, you are right. I got confused. Good catch!